jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -294,11 +334,14 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
                 uncertainties = uncert_cube.get_object(cls=StdDevUncertainty)
             flux = nddata.data << nddata.unit
             mask = nddata.mask
+            shape_mask = np.ones_like(nddata.data)
```

Comment:
You might want to demote the dtype here. Mask is just boolean, not float, right?

Reply:
this actually is supposed to be floats (they're essentially weights)